### PR TITLE
Issue 137: unify argument patterns

### LINKED ISSUE #137: Unify argument pattern in `/pipelines` for scripts aimed at loop inference/postprocessing
@dylanhmorris suggests this convention:
`filename.py positional_argument --named-argument namedvalue --other-named-argument othernamedvalue`

Which happens to line up with what I suggested, though I did not know that at the time.

_Originally posted by @damonbayer in https://github.com/CDCgov/pyrenew-hew/pull/131#discussion_r1847044727_

We should unify the pattern of arg usage in scripts.

It seems that the consensus have these features:
- Directory targets to be positional and first place.
- snake case for positional args
- kebab case for kwargs.

The changes here will be partly stylistic but also updates on the `main` and `main`-like scripts.

### PR DESCRIPTION
**Pull request by SamuelBrand1**
This draft PR ultimately aims to close #137.

## Lists of scripts

These are the scripts to bring into common pattern as described in #137. In this case "script" means a file that is designed to have its code executed with a combination of positional and named args.

I've restricted to scripts where there is an associated bash shell script. Other scripts have patterns like input and output, or seem to have positional `disease` arguments, and these are unchanged in this PR.

### Python scripts

- [x] `fit_model.py`
- [x] `generate_predictive.py`

### R scripts

- [x] `postprocess_state_forecast.R`
- [x] `score_forecast.R`

### Bash scripts

- [x] `loop_fit.sh`
- [x] `loop_generate_predictive.sh`
- [x] `loop_postprocess.sh`
- [x] `loop_score.sh`

## Testing

Where possible I've adapted the experimental data generation from #117 and #131 to check that scripts still run via shell script calls on test data. But only for functions in `loop_fit.sh` and `loop_generate_predictive.sh` so far.

### CODE REVIEW

**Comment by SamuelBrand1** [comment]
The positional form for `fit_model` script was causing an error in its call down to `build_model_from_dir` because `runpy.run_path(prior_path)` was failing 

> AttributeError: 'PosixPath' object has no attribute 'startswith'

An easy fix committed here was to convert to string at this point.

**Comment by SamuelBrand1** [comment]
Last push gave error

> Error: buildx failed with: ERROR: failed to solve: cfaprdbatchcr.azurecr.io/pyrenew-hew-dependencies:137-unify-argument-pattern-in-pipelines: failed to resolve source metadata for cfaprdbatchcr.azurecr.io/pyrenew-hew-dependencies:137-unify-argument-pattern-in-pipelines: failed to authorize: failed to fetch anonymous token: unexpected status from GET request to https://cfaprdbatchcr.azurecr.io/oauth2/token?scope=repository%3Apyrenew-hew-dependencies%3Apull&service=cfaprdbatchcr.azurecr.io: 401 Unauthorized

This seems unconnected to last commit (which fixed an error).

**Comment by SamuelBrand1** [comment]
> Last push gave error
> 
> > Error: buildx failed with: ERROR: failed to solve: cfaprdbatchcr.azurecr.io/pyrenew-hew-dependencies:137-unify-argument-pattern-in-pipelines: failed to resolve source metadata for cfaprdbatchcr.azurecr.io/pyrenew-hew-dependencies:137-unify-argument-pattern-in-pipelines: failed to authorize: failed to fetch anonymous token: unexpected status from GET request to [cfaprdbatchcr.azurecr.io/oauth2/token?scope=repository%3Apyrenew-hew-dependencies%3Apull&service=cfaprdbatchcr.azurecr.io](https://cfaprdbatchcr.azurecr.io/oauth2/token?scope=repository%3Apyrenew-hew-dependencies%3Apull&service=cfaprdbatchcr.azurecr.io): 401 Unauthorized
> 
> This seems unconnected to last commit (which fixed an error).

After some inconsequential changes to trigger CI this resolved.

@gvegayon should we expect stochastic failure in the container build CI?

**Comment by SamuelBrand1** [comment]
This should be sequenced ahead of #117 now.

File: pipelines/build_model.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def build_model_from_dir(model_dir):
         - 1
     )
 
-    priors = runpy.run_path(prior_path)
+    priors = runpy.run_path(str(prior_path))
```
Comment (dylanhmorris) [inline pipelines/build_model.py:51]: why is this needed?

Comment (dylanhmorris) [reply pipelines/build_model.py:51]: (I think this can be merged regardless, but I am curious)
